### PR TITLE
Feat: Add "data" to spans

### LINKED ISSUE #1233: Span.data is missing
https://develop.sentry.dev/sdk/event-payloads/span/

We don't have a field to set arbitrary data associated with the Span, wondering if we decided to drop it or was an oversight.

Context: https://github.com/getsentry/sentry-cocoa/pull/932#discussion_r570927177

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
## :scroll: Description

Add "data" to spans.

## :bulb: Motivation and Context

Fixes #1233

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] No breaking changes

### CODE REVIEW

File: sentry/src/main/java/io/sentry/ISpan.java [hunk 1]
```diff
@@ -154,4 +154,20 @@ ISpan startChild(
    * @return if span has finished.
    */
   boolean isFinished();
+
+  /**
+   * Sets extra data on span on transaction.
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/ISpan.java:159]:
```suggestion
   * Sets extra data on span or transaction.
```

File: sentry/src/test/java/io/sentry/SentryTracerTest.kt [hunk 2]
```diff
@@ -498,4 +498,28 @@ class SentryTracerTest {
             assertNotNull(it.value)
         }
     }
+
+    @Test
+    fun `sets ITransaction data as extra in SentryTransaction`() {
+        val transaction = fixture.getSut(sampled = true)
+        transaction.setData("key", "val")
+        transaction.finish()
+        verify(fixture.hub, times(1)).captureTransaction(check {
```
Comment (marandaneto) [inline sentry/src/test/java/io/sentry/SentryTracerTest.kt:507]:
```suggestion
        verify(fixture.hub).captureTransaction(check {
```

File: sentry/src/test/java/io/sentry/SentryTracerTest.kt [hunk 3]
```diff
@@ -498,4 +498,28 @@ class SentryTracerTest {
...
+    }
+
+    @Test
+    fun `sets Span data as data in SentrySpan`() {
+        val transaction = fixture.getSut(sampled = true)
+        val span = transaction.startChild("op")
+        span.setData("key", "val")
+        span.finish()
+        transaction.finish()
+        verify(fixture.hub, times(1)).captureTransaction(check {
```
Comment (marandaneto) [inline sentry/src/test/java/io/sentry/SentryTracerTest.kt:519]:
```suggestion
        verify(fixture.hub).captureTransaction(check {
```

**Review by marandaneto — APPROVED** [review]

**Comment by maciejwalkowiak** [comment]
Thanks for polishing @marandaneto

**Pull request merged**